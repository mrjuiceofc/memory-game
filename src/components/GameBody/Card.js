export default function Card(props) {
  const { url, handleClick, id } = props;
  return (
    <div onClick={handleClick.bind(null, id)}>
      <button className="text-6xl bg-violet-100 border border-violet-300 h-28 w-28 rounded-full hover:shadow-2xl shadow-violet-300">
        {url}
      </button>
    </div>
  );
}
