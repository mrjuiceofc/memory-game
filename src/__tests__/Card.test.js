import Card from "../components/Card.js";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const mockHandleClick = jest.fn();

describe("Card component", () => {
  it("renders Card component", () => {
    const { container } = render(
      <Card handleClick={mockHandleClick} emoji="ðŸ˜‚" />
    );
    expect(container).toMatchSnapshot();
  });

  it("calls handleClick correctly", () => {
    render(<Card handleClick={mockHandleClick} emoji="ðŸ˜‚" />);
    const button = screen.getByRole("button");
    userEvent.click(button);
    expect(mockHandleClick).toHaveBeenCalledTimes(1);
  });
});
