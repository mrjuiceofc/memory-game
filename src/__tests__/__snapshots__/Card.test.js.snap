// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Card component 1`] = `
<div>
  <button
    class="text-6xl bg-violet-100 border block border-violet-300 h-28 w-28  rounded-full hover:shadow-2xl shadow-violet-300"
  >
    ðŸ˜‚
  </button>
</div>
`;
